test/extensions/secondorder/diag_ggn/test_diag_ggn.py

```diff
@@ -43,6 +43,7 @@ def test_diag_ggn_batch(problem):
 
 
 MC_ATOL = 1e-4
+MC_BATCH_ATOL = 1e-2
```

Comment:
This value is large. I am concerned that we would not catch failures with it.

Can we at least test that the MC approximation becomes better as we use more samples? I.e. the test below with 1000 samples should require larger tolerances to pass than the test with 100.000 samples?